modules/mini-games/tightspot.lua

```diff
@@ -0,0 +1,743 @@
+        variables.level.starting_land_prize
+end
+
+local str_format = string.format
+local function SecondsToClock(seconds)
+    seconds = tonumber(seconds)
+
+    if seconds <= 0 then
+        return "00:00"
+    else
+        local mins = str_format("%02.f", math.floor(seconds / 60))
+        local secs = str_format("%02.f", math.floor(seconds - mins * 60))
+        return mins .. ":" .. secs
+    end
+end
+
+local function change_balance(player, amount)
+    if amount < 0 then
+        local available = player.get_item_count("coin")
+        amount = amount * -1
+        if available < amount then
+            local Debt = 5000 - amount
+            player.insert {name = "coin", count = Debt}
+            player.print("Borrowed 5000 coins to pay for the land")
+            local Main_gui = Gui.get_left_element(player, game_gui)
+            local gui_table = Main_gui.container["Money"].table
+            gui_table["Debt"].caption = tostring(tonumber(gui_table["Debt"].caption) + 5000)
+        else
+            player.remove_item {name = "coin", count = amount}
+        end
+    elseif amount ~= 0 then
+        player.insert {name = "coin", count = amount}
+    end
+    Store.set(balances, player, player.get_item_count("coin"))
+end
+
+local function player_join_game(player, at_player)
+    --coins and random stuffs
+    local level = variables.level
+    local character = player.character
+    player.set_controller {type = defines.controllers.god}
+    if character and character.valid then
+        character.destroy()
+    end
+    player.force.manual_mining_speed_modifier = 1000
+    Store.set(balances, player, variables.diffuclty)
+    player.insert {name = "coin", count = variables.diffuclty}
+
+    --gui
+
+    local Main_gui = Gui.get_left_element(player, game_gui)
+    Gui.toggle_left_element(player, game_gui, true)
+    local gui_table = Main_gui.container["Money"].table
+    gui_table["Balance"].caption = variables.diffuclty
+    gui_table["Time"].caption = SecondsToClock(level.time / 60)
+
+    gui_table = Main_gui.container["Objectives"].table
+    gui_table["objective"].caption = level.objective
+    gui_table["prices"].caption = level.demand.price
+
+    --island
+    local player_offset = at_player * 500
+    local level_area = level.area
+    local area = {
+        {level_area[1][1] + player_offset, level_area[1][2]},
+        {level_area[2][1] + player_offset, level_area[2][2]}
+    }
+    islands[player.name] = area
+    clean_up(area)
+    local tiles = {}
+    for i, tile in ipairs(save["tiles"]) do
+        tiles[i] = tile
+        tiles[i].position.x = tiles[i].position.x + player_offset
+    end
+    variables["surface"].set_tiles(tiles)
+    for i, entity in ipairs(save.entities) do
+        local name = entity[1]
+        local position = {x = entity[2].x + player_offset, y = entity[2].y}
+        local force = entity[3]
+        local minable = entity[4]
+        position.x = position.x + player_offset
```

Comment:
This will cause the x position to be offset twice, since line 187 already incudes the player_offset